### PR TITLE
Add variable for setting the project on cluster resource

### PR DESCRIPTION
**Pull request by thunderboltsid**
The change also includes a patch to set the project property on nutanixmachinetemplate resources for control plane and worker machine deployments.

### CODE REVIEW

File: templates/clusterclass/clusterclass.yaml [hunk 1]
```diff
@@ -263,6 +263,35 @@ spec:
...
+          matchResources:
+            machineDeploymentClass:
+              names:
+                - nutanix-quick-start-worker
+        jsonPatches:
+          - op: add
+            path: /spec/template/spec/project
+            valueFrom:
+              template: |
+                type: name
```
Comment (deepakm-ntnx) [inline templates/clusterclass/clusterclass.yaml:293]: we should support uuid as well by adding type as part of variable itself

File: templates/clusterclass/clusterclass.yaml [hunk 2]
```diff
@@ -376,3 +405,9 @@ spec:
                     enum:
                       - name
                       - uuid
+  - name: projectName
```
Comment (deepakm-ntnx) [inline templates/clusterclass/clusterclass.yaml:408]: should be project and allow specifying name and uuid both types

File: templates/testdata/cluster-with-project.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          bootType: legacy
+          clusterName: fake-cluster
+          imageName: ubuntu-2204-kube-v1.29.2.qcow2
+          memorySize: 4Gi
+          subnetName: fake-subnet
+          systemDiskSize: 40Gi
+          vcpuSockets: 2
+          vcpusPerSocket: 1
+      - name: projectName
+        value: fake-project
```
Comment (deepakm-ntnx) [inline templates/testdata/cluster-with-project.yaml:48]: pls add a test for project with uuid as well after supporting uuid as a type

File: templates/clusterclass/clusterclass.yaml (hunk 2 above)
Comment (deepakm-ntnx) [reply templates/clusterclass/clusterclass.yaml:408]: we can take uuid in all the vars wherever application in separate PR

**Review by deepakm-ntnx — APPROVED** [review]
lets add uuid support in separate PR if needed

**Review by deepakm-ntnx — APPROVED** [review]

**Comment by thunderboltsid** [comment]
> lets add uuid support in separate PR if needed

I have updated the variable to admin `project` instead of `projectName` and it now admits a type, name, and uuid instead of just a name. PTAL.

**Review by deepakm-ntnx — APPROVED** [review]

**Pull request merged**